models/aeif_psc_delta.cpp

```diff
@@ -0,0 +1,536 @@
+  const long from,
+  const long to )
+{
+  assert(
+    to >= 0 && ( delay ) from < kernel().connection_manager.get_min_delay() );
+  assert( from < to );
+  assert( State_::V_M == 0 );
+  const double h = Time::get_resolution().get_ms();
+  const double tau_m_ = P_.C_m / P_.g_L;
+  for ( long lag = from; lag < to; ++lag )
+  {
+    double t = 0.0;
+
+    // numerical integration with adaptive step size control:
+    // ------------------------------------------------------
+    // gsl_odeiv_evolve_apply performs only a single numerical
+    // integration step, starting from t and bounded by step;
+    // the while-loop ensures integration over the whole simulation
+    // step (0, step] if more than one integration step is needed due
+    // to a small integration step size;
+    // note that (t+IntegrationStep > step) leads to integration over
+    // (t, step] and afterwards setting t to step, but it does not
+    // enforce setting IntegrationStep to step-t
+    while ( t < B_.step_ )
+    {
+      const int status = gsl_odeiv_evolve_apply( B_.e_,
+        B_.c_,
+        B_.s_,
+        &B_.sys_,             // system of ODE
+        &t,                   // from t
+        B_.step_,             // to t <= step
+        &B_.IntegrationStep_, // integration step size
+        S_.y_ );              // neuronal state
+
+      if ( status != GSL_SUCCESS )
+      {
+        throw GSLSolverFailure( get_name(), status );
+      }
+      // check for unreasonable values; we allow V_M to explode
+      if ( S_.y_[ State_::V_M ] < -1e3 || S_.y_[ State_::W ] < -1e6
+        || S_.y_[ State_::W ] > 1e6 )
+      {
+        throw NumericalInstability( get_name() );
+      }
+
+      // spikes are handled inside the while-loop
+      // due to spike-driven adaptation
+      if ( S_.r_ == 0 )
+      {
+        // neuron not refractory
+        S_.y_[ State_::V_M ] =
+          S_.y_[ State_::V_M ] + B_.spikes_.get_value( lag );
+
+        // if we have accumulated spikes from refractory period,
+        // add and reset accumulator
+        if ( P_.with_refr_input_ && S_.refr_spikes_buffer_ != 0.0 )
+        {
+          S_.y_[ State_::V_M ] += S_.refr_spikes_buffer_;
+          S_.refr_spikes_buffer_ = 0.0;
+        }
+      }
+      else // neuron is absolute refractory
+      {
+        S_.y_[ State_::V_M ] = P_.V_reset_; // clamp it to V_reset
+
+        // read spikes from buffer and accumulate them, discounting
+        // for decay until end of refractory period
+        if ( P_.with_refr_input_ )
+        {
+          S_.refr_spikes_buffer_ +=
+            B_.spikes_.get_value( lag ) * std::exp( -S_.r_ * h / tau_m_ );
+        }
+        else
+        {
+          B_.spikes_.get_value( lag ); // clear buffer entry, ignore spike
+        }
+      }
+      if ( S_.r_ == 0 and S_.y_[ State_::V_M ] >= V_.V_peak_ )
+      {
+        S_.y_[ State_::V_M ] = P_.V_reset_;
+        // add 1 to compensate for count-down right after loop
```

Comment:
Replace with more detailed comment from `aeif_cond_alpha.cpp`:
```c++
        /* Initialize refractory step counter.
         * - We need to add 1 to compensate for count-down immediately after
         *   while loop.
         * - If neuron has no refractory time, set to 0 to avoid refractory
         *   artifact inside while loop.
         */

```